### ISSUE TITLE
Problem with openwrt-23.05.2

### ISSUE THREAD

**Issue by ArminCoin** [eventure/hide.client.routers#2]
I get an error in openWrt-23.05.2 when I want to add the HideMe-Wireguard Interface.

RPCError
RPC call to luci.hidemevpn/getLocations failed with error -32000: Object not found
  at handleCallReply (http://192.168.55.1/luci-static/resources/rpc.js?v=git-23.306.39416-c86c256:11:3)

![grafik](https://github.com/eventure/hide.client.routers/assets/55330150/b206a024-7045-458f-990c-d929fb31285e)

Could you please help me?

**Comment by dpensa** [comment]
Hi!

In OpenWrt 23.05 some things have changed in LuCI.

As mentioned in one [post](https://forum.openwrt.org/t/trying-to-modify-luci-no-usr-lib-lua-folder/176393/4) on the official OpenWrt forum:

> LuCI doesn't use lua by default any more. It is nowadays JavaScript and ucode based. (Since October 2022, so 23.05 is the first non-lua release)

Our script requires lua, and few other things.

Until script is updated to natively support OpenWrt 23.05, some additional installation steps are needed on this newer OpenWrt versions.

Everything is described in an updated README.md (3b58735), under section [Installation on OpenWrt 23.05](https://github.com/eventure/hide.client.routers#installation-on-openwrt-2305).

In short you need to run following commands before installing script:

```
opkg update
opkg install lua luci-lib-jsonc
mkdir -p /usr/libexec/rpcd
```

Hope this helps.

**Comment by ArminCoin** [comment]
Thank you so much for your help, it worked👍 
I've encountered another issue, this time with a DNS leak problem. I've managed to fix it by deactivating the 'Use DNS servers advertised by peer' option on the WAN port, which worked well. However, whenever I restart the VPN adapter or reboot the router, I'm unable to connect to the VPN because there's no DNS to resolve the VPN connection domain. Do you happen to have a solution or any advice for this issue? I'd greatly appreciate your help!

**Comment by dpensa** [comment]
Glad that it worked!

Regarding DNS issue, I recommend that you check on [OpenWrt's forum](https://forum.openwrt.org/), as there are quite a few different solutions, depending on your use case.
